common/src/main/java/org/opensearch/ml/common/MLModelGroup.java

```diff
@@ -0,0 +1,206 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.ml.common;
+
+import lombok.Builder;
+import lombok.Getter;
+import lombok.Setter;
+import org.opensearch.common.io.stream.StreamInput;
+import org.opensearch.common.io.stream.StreamOutput;
+import org.opensearch.commons.authuser.User;
+import org.opensearch.core.xcontent.ToXContentObject;
+import org.opensearch.core.xcontent.XContentBuilder;
+import org.opensearch.core.xcontent.XContentParser;
+
+import java.io.IOException;
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.opensearch.common.xcontent.XContentParserUtils.ensureExpectedToken;
+
+@Getter
+public class MLModelGroup implements ToXContentObject {
+    public static final String MODEL_GROUP_NAME_FIELD = "name"; //name of the model group
+    // We use int type for version in first release 1.3. In 2.4, we changed to
+    // use String type for version. Keep this old version field for old models.
+    public static final String DESCRIPTION_FIELD = "description"; //description of the model group
+    public static final String LATEST_VERSION_FIELD = "latest_version"; //latest model version added to the model group
+    public static final String BACKEND_ROLES_FIELD = "backend_roles"; //back_end roles as specified by the owner/admin
+    public static final String OWNER = "owner"; //user who creates/owns the model group
+
+    public static final String ACCESS = "access"; //assigned to public, private, or null when model group created
```

Comment:
It's ok to use String here, but better to use the enum class `ModelAccessMode`.

Reply:
yeah makes sense. Will change it later